ui/redux/actions/content.js

```diff
@@ -298,17 +298,25 @@ export function doToggleLoopList(collectionId: string, loop: boolean, hideToast:
   };
 }
 
-export function doToggleShuffleList(collectionId: string, shuffle: boolean, hideToast: boolean) {
+export function doToggleShuffleList(currentUri: string, collectionId: string, shuffle: boolean, hideToast: boolean) {
   return (dispatch: Dispatch, getState: () => any) => {
     if (shuffle) {
       const state = getState();
       const urls = makeSelectUrlsForCollectionId(collectionId)(state);
 
-      const newUrls = urls
+      let newUrls = urls
```

Comment:
Is it worth filtering out at least the top 5 in content/history reducer? Or even the top <collection_size>?